ksqldb-api-client/src/main/java/io/confluent/ksql/api/client/impl/ExecuteQueryResponseHandler.java

```diff
@@ -59,30 +62,32 @@
 
   @Override
   protected void handleMetadata(final QueryResponseMetadata queryResponseMetadata) {
-    cf.queryID().complete(queryResponseMetadata.queryId);
+    final String queryId = queryResponseMetadata.queryId;
+
+    cf.queryID().complete(queryId);
+
     columnNames = queryResponseMetadata.columnNames;
-    columnTypes = RowUtil.columnTypesFromStrings(queryResponseMetadata.columnTypes);
+
+    final List<String> columnTypesString = queryResponseMetadata.columnTypes;
+
+    columnTypes = RowUtil.columnTypesFromStrings(columnTypesString);
+
     columnNameToIndex = RowUtil.valueToIndexMap(columnNames);
   }
 
   @Override
-  protected void handleRow(final Buffer buff) {
-    final Row row;
-    final Object json = buff.toJson();
-
-    if (json instanceof JsonObject) {
-      final JsonObject jsonObject = (JsonObject) json;
+  protected void handleRow(final StreamedRow row) {
+    if (row.getConsistencyToken().isPresent()) {
       // This is the serialized consistency vector
       // Don't add it to the result list since the user should not see it
-      if (jsonObject.getMap() != null && jsonObject.getMap().containsKey("consistencyToken")) {
-        log.info("Response contains consistency vector " + jsonObject);
-        serializedConsistencyVector.set((String) ((JsonObject) json).getMap().get(
-            "consistencyToken"));
-      } else {
-        throw new RuntimeException("Could not decode JSON, expected consistency toke: " + json);
-      }
-    } else  if (json instanceof JsonArray) {
-      final JsonArray values = new JsonArray(buff);
+      final ConsistencyToken consistencyToken = row.getConsistencyToken().get();
+      final String consistencyTokenString = consistencyToken.getConsistencyToken();
+      log.info("Response contains consistency vector " + consistencyTokenString);
+      serializedConsistencyVector.set(consistencyTokenString);
+    } else  if (row.getRow().isPresent()) {
+      final DataRow dataRow = row.getRow().get();
+      final Gson gson = new GsonBuilder().serializeNulls().create();
+      final JsonArray values = new JsonArray(gson.toJson(dataRow.getColumns()));
```

Comment:
We parse using Jackson into StreamedRow only to stringify it again, and then reparse it into Vertex json object?  That seems like it should be avoidable.   

We should either switch to using pojos everywhere rather than the Vertex json, in which case there's no need for this double parsing, or if it's more trouble than it's worth, skip parsing into the StreamedRow object with jackson. You would still request `application/vnd.ksql.v1+json`, but do `final Object json = buff.toJson();` as before and ask for fields by name.  Instead of it being the delimited format, it's the `application/vnd.ksql.v1+json` format, so the code is similar, but just is looking for slightly different fields.  This would still accomplish the main goal of just moving to the common format.  Tell me what you think.

Reply:
using `readTree` now instead to extract the fields